assets/src/components/form/inputs/radio-input.js

```diff
@@ -14,100 +14,87 @@ import { OptionLabelSize } from './base/option-label-size';
 /**
  * generates one or more html radio inputs
  *
- * @constructor
- * @param {string} htmlId
- * @param {string} htmlClass
- * @param {string} value
- * @param {Array} options
- * @param {Function} onChange
- * @param {boolean} btnGroup
- * @param {string} helpTextID
- * @param {Object} dataSet
- * @param {string} inputWidth
- * @param {Object} attributes
- * @return {string} rendered date name form row
+ * @function
+ * @param {Object} props
+ * @return {string} rendered radio inputs
  */
-export class RadioInput extends Component {
-	static propTypes = {
-		htmlId: PropTypes.string.isRequired,
-		htmlClass: PropTypes.string,
-		value: PropTypes.oneOfType( [
-			PropTypes.number,
-			PropTypes.string,
-		] ),
-		options: PropTypes.array,
-		onChange: PropTypes.func,
-		btnGroup: PropTypes.bool,
-		inputWidth: PropTypes.string,
-		helpTextID: PropTypes.string,
-		dataSet: PropTypes.object,
-	};
+export const RadioInput = ( {
+	name,
+	htmlId,
+	htmlClass,
+	options,
+	helpTextID,
+	btnGroup = true,
+	inputWidth = '',
+	...attributes
+} ) => {
+	const [ checked, setChecked ] = useState( setCheckedState() );
 
-	constructor( props ) {
-		super( props );
-		this.state = {
-			checked: OptionCheckedState(
-				props.htmlId,
-				props.options,
-				props.initialValue,
-			),
-		};
-	}
+	const labelClass = btnGroup ?
+		'btn btn-primary' :
+		'ee-radio-label-after' + OptionLabelSize( options );
+	let divClass = btnGroup ? 'btn-group' : 'ee-checkbox-group';
+	divClass = inputWidth ?
+		`${ divClass } ee-input-width-${ inputWidth }` :
+		divClass;
 
+	/**
+	 * @function
+	 * @param {mixed} reset
+	 * @return {Object} state
+	 */
+	const setCheckedState = ( reset ) => {
+		return OptionCheckedState(
+			htmlId,
+			options,
+			attributes.initialValue,
```

Comment:
same comment as above regarding whether this should be converted to a prop...
I'm guessing this was done the way it was originally to avoid having to manage that prop within the class in other ways (ie: dumping into state, etc), but now that this is converted to a functional component, it's trivial to bring it in as its own prop and access it from anywhere.